ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_2d_program_factory.cpp

```diff
@@ -89,37 +91,52 @@ operation::ProgramWithCallbacks create_program_mcast_in0_in1(
     const bool in1_is_sharded = in1_buffer->buffer_layout() == TensorMemoryLayout::WIDTH_SHARDED;
     const bool output_is_sharded = out_buffer->buffer_layout() == TensorMemoryLayout::BLOCK_SHARDED;
 
-    uint32_t in0_block_tiles = per_core_M * in0_block_w;
+    bool do_not_inplace_interm0_out_CB = output_is_sharded && (per_core_M != out_block_h);
+
+    // hardcode testing, testing multiple blocks on height for now
+    uint32_t in0_block_h = out_block_h;
```

Comment:
will you remove this code once ready to merge? question based on the comment

Reply:
removed outdated comment